extension/src/setup/studio.ts

```diff
@@ -0,0 +1,129 @@
+import { EventEmitter } from 'vscode'
+import { isStudioAccessToken } from './token'
+import { AvailableCommands, InternalCommands } from '../commands/internal'
+import { getFirstWorkspaceFolder } from '../vscode/workspaceFolders'
+import { Args, ConfigKey, Flag } from '../cli/dvc/constants'
+import { ContextKey, setContextValue } from '../vscode/context'
+
+export class Studio {
+  protected studioConnectionChanged: EventEmitter<void>
+  private studioAccessToken: string | undefined = undefined
+  private studioIsConnected = false
+  private shareLiveToStudio: boolean | undefined = undefined
+  private readonly getCwd: () => string | undefined
+  private readonly internalCommands: InternalCommands
```

Comment:
This class would be handling token requests and responses when we implement Studio token authentication.